src/Symfony/Component/HttpKernel/DependencyInjection/ServicesResetter.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\DependencyInjection;
+
+/**
+ * Resets provided services.
+ *
+ * @author Alexander M. Turek <me@derrabus.de>
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+class ServicesResetter
+{
+    private $resettableServices;
+    private $resetMethods;
+
+    public function __construct(\Traversable $resettableServices, array $resetMethods)
```

Comment:
How about having only one argument with an iterator over an array of callables, so that instead of `$service->{$this->resetMethods[$id]}();` you'll do `$callable()`, where `$callable` is `[$service, $resetMethod]`

Reply:
the iterators build by the container using IteratorArgument are restricted to yielding services only, so that's not possible with the current primitives (and we don't really care, so all is fine :) )